test/TestApps/ToastNotificationsTestApp/main.cpp

```diff
@@ -510,9 +510,195 @@ bool VerifyShowToast_Unpackaged()
+
+    winrt::ToastNotification toast2{ CreateToastNotification() };
+    toast2.Tag(L"Shared tag");
+    toast2.Group(L"group2");
+    toastNotificationManager.ShowToast(toast2);
+
+    winrt::ToastNotification toast3{ CreateToastNotification() };
+    toast3.Tag(L"Shared tag");
+    toast3.Group(L"group3");
+    toastNotificationManager.ShowToast(toast3);
+
+    if (!VerifyToastIsActive(toast1.ToastId()) || !VerifyToastIsActive(toast2.ToastId()) || !VerifyToastIsActive(toast3.ToastId()))
+    {
+        return false;
+    }
+
+    auto removeToastAsync{ winrt::ToastNotificationManager::Default().RemoveWithTagAsync(L"Shared tag") };
+    if (removeToastAsync.wait_for(std::chrono::seconds(300)) != winrt::Windows::Foundation::AsyncStatus::Completed)
+    {
+        return false;
+    }
+
+    if (VerifyToastIsActive(toast1.ToastId()) || VerifyToastIsActive(toast2.ToastId()) || VerifyToastIsActive(toast3.ToastId()))
+    {
+        return false;
+    }
+
+    return true;
+}
+
+bool VerifyRemoveWithTagGroupAsync()
+{
+    auto toastNotificationManager = winrt::ToastNotificationManager::Default();
+
+    winrt::ToastNotification toast1{ CreateToastNotification() };
+    toast1.Tag(L"tag1");
+    toast1.Group(L"Shared group");
+    toastNotificationManager.ShowToast(toast1);
+
+    winrt::ToastNotification toast2{ CreateToastNotification() };
+    toast2.Tag(L"tag2");
+    toast2.Group(L"Shared group");
+    toastNotificationManager.ShowToast(toast2);
+
+    winrt::ToastNotification toast3{ CreateToastNotification() };
+    toast3.Tag(L"tag3");
+    toast3.Group(L"Shared group");
+    toastNotificationManager.ShowToast(toast3);
+
+    if (!VerifyToastIsActive(toast1.ToastId()) || !VerifyToastIsActive(toast2.ToastId()) || !VerifyToastIsActive(toast3.ToastId()))
+    {
+        return false;
+    }
+
+    auto removeToastAsync{ winrt::ToastNotificationManager::Default().RemoveWithTagGroupAsync(L"tag2", L"Shared group")};
+    if (removeToastAsync.wait_for(std::chrono::seconds(300)) != winrt::Windows::Foundation::AsyncStatus::Completed)
+    {
+        return false;
+    }
+
+    if (!VerifyToastIsActive(toast1.ToastId()) || VerifyToastIsActive(toast2.ToastId()) || !VerifyToastIsActive(toast3.ToastId()))
+    {
+        return false;
+    }
 
     return true;
 }
+
+bool VerifyRemoveWithGroupAsync()
+{
+    auto toastNotificationManager = winrt::ToastNotificationManager::Default();
+
+    winrt::ToastNotification toast1{ CreateToastNotification() };
+    toast1.Tag(L"tag1");
+    toast1.Group(L"Shared group");
+    toastNotificationManager.ShowToast(toast1);
+
+    winrt::ToastNotification toast2{ CreateToastNotification() };
+    toast2.Tag(L"tag2");
+    toast2.Group(L"Shared group");
+    toastNotificationManager.ShowToast(toast2);
+
+    winrt::ToastNotification toast3{ CreateToastNotification() };
+    toast3.Tag(L"tag3");
+    toast3.Group(L"Shared group");
+    toastNotificationManager.ShowToast(toast3);
+
```

Comment:
looks like all toasts are in the same group. maybe we can add another toast with a different group and verify it will not be removed.